kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storage/StorageWriteApiBase.java

```diff
@@ -297,6 +296,22 @@ protected BigQueryWriteClient getWriteClient() throws IOException {
     return this.writeClient;
   }
 
+  /**
+   * Returns a {@link JsonStreamWriterFactory} for creating configured {@link JsonStreamWriter} instances
+   *
+   * @return a {@link JsonStreamWriterFactory}
+   */
+  protected JsonStreamWriterFactory getJsonWriterFactory() {
+    RetrySettings retrySettings = RetrySettings.newBuilder()
+            .setMaxAttempts(retry)
+            .setInitialRetryDelay(Duration.ofMillis(retryWait))
+            .setMaxRetryDelay(Duration.ofMillis(retryWait))
```

Comment:
This should be a larger value, that encompasses the time after all retry attempts have been made.